src/data/models.sql

```diff
@@ -0,0 +1,296 @@
+);
+
+CREATE TABLE `dayPass` (
+	`id` INT NOT NULL AUTO_INCREMENT,
+	`studentId` varchar(10) NOT NULL,
+	`date` DATE NOT NULL,
+	`reason` varchar,
+	`consentType` varchar,
+	`approvedBy` varchar,
+	`comment` varchar,
+	PRIMARY KEY (`id`)
+);
+
+CREATE TABLE `loginType` (
+	`id` INT(2) NOT NULL AUTO_INCREMENT,
+	`type` varchar NOT NULL UNIQUE,
+	PRIMARY KEY (`id`)
+);
+
+CREATE TABLE `ps` (
+	`id` varchar(10) NOT NULL,
+	`psStation` varchar,
+	`thesis` BOOLEAN NOT NULL,
+	PRIMARY KEY (`id`)
+);
+
+CREATE TABLE `merchandiseOption` (
+	`id` INT NOT NULL AUTO_INCREMENT,
+	`size` varchar NOT NULL,
+	`nick` varchar,
+	PRIMARY KEY (`id`)
+);
+
+CREATE TABLE `nucleus` (
+	`facultyId` INT NOT NULL,
+	PRIMARY KEY (`facultyId`)
+);
+
+CREATE TABLE `staff` (
+	`loginId` varchar NOT NULL,
+	`name` varchar NOT NULL,
+	`type` varchar NOT NULL,
+	`phone` varchar NOT NULL,
+	PRIMARY KEY (`loginId`)
+);
+
+CREATE TABLE `messOption` (
+	`studentId` varchar(10) NOT NULL,
+	`monthYear` varchar NOT NULL,
+	`mess` varchar,
+	PRIMARY KEY (`studentId`,`monthYear`)
+);
+
+CREATE TABLE `hostel` (
+	`id` varchar(10) NOT NULL,
+	`hostel` varchar NOT NULL,
+	`room` varchar NOT NULL,
+	PRIMARY KEY (`id`)
+);
+
+ALTER TABLE `student` ADD CONSTRAINT `student_fk0` FOREIGN KEY (`id`) REFERENCES `loginId`(`loginId`);
+
+ALTER TABLE `bonafide` ADD CONSTRAINT `bonafide_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `csa` ADD CONSTRAINT `csa_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `dayScholar` ADD CONSTRAINT `dayScholar_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `leave` ADD CONSTRAINT `leave_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `disco` ADD CONSTRAINT `disco_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `facultyIncharge` ADD CONSTRAINT `facultyIncharge_fk0` FOREIGN KEY (`facultyId`) REFERENCES `faculty`(`loginId`);
+
+ALTER TABLE `inOutDetail` ADD CONSTRAINT `inOutDetail_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `lateComer` ADD CONSTRAINT `lateComer_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `loginId` ADD CONSTRAINT `loginId_fk0` FOREIGN KEY (`type`) REFERENCES `loginType`(`id`);
+
+ALTER TABLE `notice` ADD CONSTRAINT `notice_fk0` FOREIGN KEY (`categoryId`) REFERENCES `noticeCategories`(`id`);
+
+ALTER TABLE `faculty` ADD CONSTRAINT `faculty_fk0` FOREIGN KEY (`loginId`) REFERENCES `loginId`(`loginId`);
+
+ALTER TABLE `transaction` ADD CONSTRAINT `transaction_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
+
+ALTER TABLE `transaction` ADD CONSTRAINT `transaction_fk1` FOREIGN KEY (`merchId`) REFERENCES `merchandise`(`id`);
+
+ALTER TABLE `transaction` ADD CONSTRAINT `transaction_fk2` FOREIGN KEY (`optionId`) REFERENCES `merchandiseOption`(`id`);
+
+ALTER TABLE `messBill` ADD CONSTRAINT `messBill_fk0` FOREIGN KEY (`studentId`) REFERENCES `student`(`id`);
```

Comment:
This line is giving me an error too.
```
ERROR 1215 (HY000) at line 272: Cannot add foreign key constraint
````

Reply:
there was a conflicting variable type.. i am fixing it